WordPress/gutenberg
# Block Navigation: try using CSS for indentation with known max indent instead of spacer divs

In the Block Navigation component, I noticed that we were using spacer divs instead of CSS rules. For a deeply nested child, we'd see X divs for the necessary indentation at that level.

Some blocks allow unbounded nesting, like the columns block. Proposed changes here would remove the need for an excessive number of whitespace divs. I've arbitrarily picked 8 as the max indentation level, but I'd be open to other limits/approaches.

What's a little frustrating is that [attr()](https://developer.mozilla.org/en-US/docs/Web/CSS/attr()) doesn't have support outside of content, but maybe in the future this can be done simply as a calc + attr rule.

| before | after |
|-----|-----|
| <img width="1626" alt="Screen Shot 2021-05-20 at 9 33 15 AM" src="https://user-images.githubusercontent.com/1270189/119022197-335ca380-b955-11eb-8720-a46bf54eed27.png"> | <img width="1491" alt="Screen Shot 2021-05-20 at 10 21 33 AM" src="https://user-images.githubusercontent.com/1270189/119022244-42435600-b955-11eb-93cd-404081816336.png"> |

#### Testing Instructions

* Checkout this branch
* Activate a theme that support full site editing, like TT1 blocks
* Visit the Site Editor at `/wp-admin/admin.php?page=gutenberg-edit-site`
* Click on list view in the top toolbar. (The icon to the right of the redo button).
* Try adding heavily nested blocks, like column or group. Add enough so we go past 9 levels.
* Verify that there are no regressions for levels up to 9 deep. Test and see if having a max indent level still feels usable. (Previously we could grow the size of the left panel arbitrarily large, making the content pane in the center shrink)
